pennylane/transforms/defer_measurements.py

```diff
@@ -101,9 +106,299 @@ def null_postprocessing(results):
     return results[0]
 
 
-@transform
+@lru_cache
+def _get_plxpr_defer_measurements():
+    try:
+        # pylint: disable=import-outside-toplevel
+        import jax
+
+        from pennylane.capture import PlxprInterpreter
+        from pennylane.capture.primitives import cond_prim, ctrl_transform_prim, measure_prim
+    except ImportError:  # pragma: no cover
+        return None, None
+
+    # pylint: disable=redefined-outer-name
+
+    class DeferMeasurementsInterpreter(PlxprInterpreter):
+        """Interpreter for applying the defer_measurements transform to plxpr."""
+
+        # pylint: disable=unnecessary-lambda-assignment,attribute-defined-outside-init,no-self-use
+
+        def __init__(self, aux_wires):
+            super().__init__()
+            self._aux_wires = Wires(aux_wires)
+
+            # We use a dict here instead of a normal int variable because we want the state to mutate
+            # when we interpret higher-order primitives
```

Comment:
Will this even work with higher order primitives?

Reply:
Yes, I think you're correct about this only working with `cond`, but only if the predicates do not use MCMs, since `cond` with MCMs will change to `ctrl`, and `ctrl(measure)` is not valid

Reply:
I added it to the "What doesn't work" section of the usage details I added for `defer_measurements`